nomad/structs/config/tls.go

```diff
@@ -33,6 +40,24 @@ type TLSConfig struct {
 	VerifyHTTPSClient bool `mapstructure:"verify_https_client"`
 }
 
+type KeyLoader struct {
+	Certificate *tls.Certificate
+}
+
+func (k *KeyLoader) LoadKeyPair(certFile, keyFile string) (*tls.Certificate, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load cert/key pair: %v", err)
```

Comment:
Shouldn't k.Certificate be set to nil and the load error captured in a variable so that GetOutGoingCertificate starts to return nil after that? The error case I am thinking of is when you load a cert successfully, use it for a while, then need to reload because it expired and the new cert fails to load. In that case, if I understand this code correctly, its going to return the old cert in `GetOutgoingCertificate`

Reply:
Great point. This also made me realize that we don't handle downgrading with this either (if that is something we want to handle), it is always assumed that the reload is either upgrading to TLS or changing certificates.

Reply:
Oh ya, we need to handle downgrades too.